docs/bazel/status.md

```diff
@@ -0,0 +1,339 @@
+  * Cross-platform builds: build/exec host == Linux, MacOS; target host == Linux x86_64
+
+[libsnark](https://github.com/scipr-lab/libsnark) gets special treatment, since it is central to Coda.
+Coda vendors this library - embeds a copy of it within the Coda
+codebase
+([src/lib/snarky/src/camlsnark_c/libsnark-caml/libsnark](src/lib/snarky/src/camlsnark_c/libsnark-caml/libsnark)),
+with some local modifications.
+
+Several strategies for Bazelization of libsnark are available:
+
+* Add direct (embedded) Bazel support to libsnark and all or some of its dependencies; this entails one of:
+  * Convince the upstream maintainers to incorporate Bazel support; or
+  * Maintain the Bazelized forks in the Coda repo; or
+  * Incorporate the Bazel code into the vendored image of libsnark within Coda.
+* Import libsnark as an external repo and write a local BUILD file for it.
+
+This implies a choice:
+
+* Retain the current vendored image and add Bazel support to it.
+* Treat libsnark as an external repo, and patch/augment it for Coda purposes
+  (instead of vendoring the code)
+
+I've chosen to fork and Bazelize libsnark and all of its dependencies
+(see details in [Status](#status)).  The resulting Bazel code can be
+adapted to the needs of the Coda project.
+
+### OCaml Tasks <a name="tasks_ocaml"></a>
+
+### Go Tasks <a name="tasks_go"></a>
+
+### Rust <a name="tasks_rust"></a>
+
+### Reason/Javascript <a name="tasks_js"></a>
+
+### Integration <a name="tasks_integration"></a>
+
+
+
+## Status <a name="status"></a>
+
+As of Fri 6/12/2020
+
+### C/C++ Status
+
+* Manifest of C/C++ deps: see WORKSPACE file.
+* The following libs have been Bazelized:
+  * boost
+  * libffi
+  * libgmp
+  * gtest
+  * libjemalloc
+  * libomp (OpenMP)
+  * libre2
+  * libsnark and its dependencies (details below)
+  * libsodium
+  * libssl, libcrypto (OpenSSL)
+  * zlib
+
+Remaining: rocksdb
+
+See WORKSPACE and BUILD.bazel for details.
+
+#### libsnark
+
+libsnark and its dependencies have been forked and Bazilized:
+
+* [libsnark](https://github.com/mobileink/libsnark)
+* [libfqfft](https://github.com/mobileink/libfqfft)
+* [libff](https://github.com/mobileink/libff)
+* [ate-pairing](https://github.com/mobileink/ate-pairing)
+* [xbyak](https://github.com/mobileink/xbyak)
+
+#### C/C++ Issues
+
+* libpq-dev (Postgresql)
+* libprocps-dev
+* libsnark - several targets fail to compile, with an error that does
+  not appear to be related to the build logic.  Possible causes:
+  versioning issues; MacOS eccentricities; compile/link flags. Or
+  maybe those failures are expected.  To be investigated.
+
+### OCaml Status
+
+In progress; see notes below in Schedule section.
+
+General Bazel support for OCaml is under development at
+https://github.com/mobileink/obazl.  For notes on Bazel support for
+Coda, see [Building Coda with Bazel](docs/building/bazel.md).
+
+### Go Status
+
+First cut done.  See build targets in `src/app/libp2p_helper`
```

Comment:
I believe `libp2p_helper` builds independently of all other code currently. Can we convert the `Makefile` to use bazel? It seems like this might be a viable Trojan horse to get developers set up with and using bazel.

/cc @emberian

Reply:
Yep.  In fact I just submitted PR  #5262 for this.